internal/model/modelindexer/indexer.go

```diff
@@ -434,6 +397,72 @@ func (i *Indexer) flush(ctx context.Context, bulkIndexer *bulkIndexer) error {
 	return nil
 }
 
+// runActiveIndexer starts a new active indexer which pulls items from the
+// bulkItems channel. The more active indexers there are, the faster events
+// will be pulled out of the queue, but also the more likely it is that the
+// outgoing Elasticsearch bulk requests are flushed due to the idle timer,
+// rather than due to being full.
+func (i *Indexer) runActiveIndexer() {
+	i.g.Go(func() error {
+		atomic.AddInt64(&i.activeBulkRequests, 1)
+		var mu sync.Mutex
+		// `active` and `timer` must be accessed with `mu` locked.
+		var active *bulkIndexer
+		var timer *time.Timer
+		flush := make(chan *bulkIndexer)
```

Comment:
I initially gave it a buffer of 1 but removed it since it didn't seem to make much of a difference, I might re-introduce it and micro-benchmark and see if there's any meaningful difference